hamilton/plugins/h_polars.py

```diff
@@ -54,3 +67,152 @@ def build_result(
 
     def output_type(self) -> Type:
         return pl.DataFrame
+
+
+# Do we need this here?
+class with_columns(with_columns_factory):
```

Comment:
So yeah, I think this should live individually -- the contract shouldb e per-dataframe-type (for now). Then we can add a central one, but this enables separate implementations and the fact that we're using the same class to be an implementation detail.